### DISCUSSION TITLE
File store API calls are failing at Zuul

### DISCUSSION THREAD

**Discussion by rishabhsuriwtc** [egovernments/Digit-Core#55] [DIGIT Platform]
The current limit rate in the zuul properties is 5 requests per 60 seconds and the filestore[0].type[1] has been set to origin which was present by default.

When i am trying to hit the API via my application or postman, it is failing. there are around 5 failed calls (with error code 429) made to the filestore in the zuul logs.

Why are the file store calls failing with the default zuul properties for it? Also, please note that we are using the S3 bucket for the MINIO_URL property.

- name: MINIO_URL
    value:

**Comment by rishabhsuriwtc** [comment]
any immediate help on this very urgent issue will be highly appreciated

**Reply by kavi-egov** [reply]
Hi, can you please share the curl for which the call is failing. The MINIO_URL variable is used for both AWS and MINIO, changing the url will change the source. Also provide some info on the time frame in which the calls are made, did all the calls fail or only some failed ?

**Comment by richa1502** [comment]
Hi Kavi,

The curl which gave 429 error is as :

curl 'http://175.101.1.86/filestore/v1/files/url?tenantId=mp.health&fileStoreIds=06ab9d92-dd91-4183-af5f-a8659f59c1da' \
  -H 'Accept: application/json, text/plain, /' \
  -H 'Accept-Language: en-US,en;q=0.9' \
  -H 'Connection: keep-alive' \
  -H 'Referer: http://175.101.1.86/digit-ui/employee/pgr/grievance/details/PG-PGR-2023-04-06-001632' \
  -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/[111.0.0.0](http://111.0.0.0/) Safari/537.36' \
  --compressed \
  --insecure


The approach we have followed - We changed the limit up to 100 for the same URL in the limiter.properties file in Zuul. The screenshot for the same is attached below:

<img width="551" alt="image (11)" src="https://user-images.githubusercontent.com/115135810/233358432-5e6e95f8-f639-4028-bec0-1a07847b6354.png">
The screenshot of the response on Postman is attached below : 
![image (1) (2)](https://user-images.githubusercontent.com/115135810/233359186-369c56b9-73f3-4486-b4f2-579e9a56f913.png)
